### PR TITLE
Displaying Booking reference on the MI report

### PR DESCRIPTION
**Pull request by KLV96**
**Jira Ticket:** 
https://cshrdigitalandanalysis.atlassian.net/browse/LC-280
https://cshrdigitalandanalysis.atlassian.net/browse/LC-309

This branch adds booking reference, event location, event ID and event date. 

This branch is relying on these two branches: 
https://github.com/Civil-Service-Human-Resources/lpg-learner-record/pull/107
https://github.com/Civil-Service-Human-Resources/lpg-learning-catalogue/pull/121

### CODE REVIEW

File: src/main/java/uk/gov/cshr/report/reports/BookingReportRow.java [hunk 1]
```diff
@@ -199,7 +203,40 @@ public void setPoNumber(String poNumber) {
         this.poNumber = poNumber;
     }
 
-    public boolean isPaidFor() { return paidFor; }
+    public boolean isPaidFor() {
```
Comment (kameshganesan) [inline src/main/java/uk/gov/cshr/report/reports/BookingReportRow.java]: We could leverage the use of Lombok getters and setters here.

File: src/main/java/uk/gov/cshr/report/reports/ModuleReportRow.java [hunk 2]
```diff
@@ -167,4 +167,5 @@ public String getCompletedAt() {
     public void setCompletedAt(String completedAt) {
         this.completedAt = completedAt;
     }
+
```
Comment (kameshganesan) [inline src/main/java/uk/gov/cshr/report/reports/ModuleReportRow.java]: Unnecessary newline here.

**Comment by kameshganesan** [comment]
Please resolve the conflicts.

**Comment by kameshganesan** [comment]
Please resolve conflicts.

File: src/main/java/uk/gov/cshr/report/factory/ReportRowFactory.java [hunk 3]
```diff
@@ -41,11 +41,17 @@ public BookingReportRow createBookingReportRow(Optional<CivilServant> civilServa
             reportRow.setModuleId(event.getModule().getId());
             reportRow.setModuleTitle(event.getModule().getTitle());
             reportRow.setRequired(event.getModule().getRequired());
-            reportRow.setPaidFor(event.getModule().getAssociatedLearning());
+            if(event.getModule().getAssociatedLearning() != null) {
```
Comment (kameshganesan) [inline src/main/java/uk/gov/cshr/report/factory/ReportRowFactory.java]: space after 'if'

File: src/main/java/uk/gov/cshr/report/reports/BookingReportRow.java [hunk 4]
```diff
@@ -30,176 +33,8 @@
     private String bookingCancellationReason;
     private String topicId;
     private String poNumber;
-
```
Comment (m24murray) [inline src/main/java/uk/gov/cshr/report/reports/BookingReportRow.java:33]: @KLV96 @kameshganesan

We actually removed lombok from this class as it was causing some issues in the past with the ordering of data on the csv. See the following commit:
https://github.com/Civil-Service-Human-Resources/lpg-report-service/commit/354d7c17f73fc4c26b648faf376783d1b468c130#diff-dc9c89b6d759619c69abb8ffed04deb4

Can you reinstate please?

Comment (KLV96) [reply src/main/java/uk/gov/cshr/report/reports/BookingReportRow.java:33]: done

**Pull request merged**